ansible/validate.yml

```
@@ -43,6 +43,34 @@
         # below produced by dev/setup-env.sh - gives empty list if file is missing:
         _requirements_installed: "{{ ((lookup('file', _requirements_path + '.last', errors='ignore') or '{}') | from_yaml ).values() | flatten }}"
 
+- name: Validate OpenTofu templated inventory is appropriate
+  # This "documents" the assumptions that Ansible makes about the
+  # OpenTofu-provided inventory
+  hosts: localhost
+  gather_facts: false
+  tags:
+    - validate
+    - opentofu
+  tasks:
+    - name: Check templated groups
+      assert:
+        that:
+          - item in groups
+          - groups[item] | length > 0
+        fail_msg: "Expected inventory group '{{ item }}' is missing or empty: is OpenTofu inventory template up to date?"
+      loop:
+        - control
+        - compute
+        - login
+    - name: Check templated 'all' vars
+      assert:
+        that:
+          - openhpc_cluster_name is defined
+          - cluster_domain_suffix is defined
+          - cluster_home_volume is defined
+          - cluster_compute_groups is defined
```

do we also want to check for
```suggestion
          - cluster_compute_groups is defined
          - hostvars[groups['control'] | first].appliances_state_dir is defined
```

---

Well, theoretically its OK to run without that, in that the ansible copes ... but I agree it isn't sensible. I think I want to say that is out of scope for the moment!